### ISSUE TITLE
[Bug]: A11y addon panel does not update with new Storybook Test results

### ISSUE THREAD

**Issue by kylegach** [storybookjs/storybook#31412]
### Describe the bug

After running component tests via the testing widget, I would expect the a11y addon panel to update its results. Currently, it only updates after either manually triggering a rerun in the panel or navigating to a different story and back.

The most ideal fix would somehow update the panel automatically. Perhaps by listening to a "tests finished" event and re-running the checks?

### Reproduction link

https://github.com/yannbf/mealdrop/compare/main...retreat-demo

### Reproduction steps

1. Checkout reproduction branch
2. `yarn`
3. `yarn storybook`
4. Have a11y errors
5. Click on a11y error in a story's menu
6. A11y panel correctly shows violation
7. Navigate to different addon panel
8. Fix a11y error and rerun test
9. Click on a11y status in same story's menu
10. A11y panel still shows violation

### System

```bash
System:
    OS: macOS 15.3.2
    CPU: (8) arm64 Apple M1 Pro
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.16.0 - /private/var/folders/q7/9dmtlkxs0lxfv6nr64f86pth0000gn/T/xfs-787b3a4d/node
    Yarn: 4.2.1 - /private/var/folders/q7/9dmtlkxs0lxfv6nr64f86pth0000gn/T/xfs-787b3a4d/yarn <----- active
    npm: 10.9.2 - ~/projects/mealdrop/node_modules/.bin/npm
  Browsers:
    Chrome: 135.0.7049.117
    Safari: 18.3.1
  npmPackages:
    @storybook/addon-a11y: 9.0.0-beta.11 => 9.0.0-beta.11
    @storybook/addon-designs: 9.0.0-next.3 => 9.0.0-next.3
    @storybook/addon-docs: 9.0.0-beta.11 => 9.0.0-beta.11
    @storybook/addon-themes: 9.0.0-beta.11 => 9.0.0-beta.11
    @storybook/addon-vitest: 9.0.0-beta.11 => 9.0.0-beta.11
    @storybook/react-vite: 9.0.0-beta.11 => 9.0.0-beta.11
    @storybook/test-runner: 0.23.0-next.1 => 0.23.0-next.1
    chromatic: ^11.26.1 => 11.28.2
    eslint-plugin-storybook: 9.0.0-beta.11 => 9.0.0-beta.11
    msw-storybook-addon: ^2.0.4 => 2.0.4
    storybook: 9.0.0-beta.11 => 9.0.0-beta.11
    storybook-addon-test-codegen: patch:storybook-addon-test-codegen@npm%3A1.2.0#~/.yarn/patches/storybook-addon-test-codegen-npm-1.2.0-9550d5ad4c.patch => 1.2.0
```

### Additional context

I tried to use `storybook.new` to create a full reproduction, but ran into two issues:

1. `npx storybook add @storybook/addon-a11y` installed the package but the postinstall script either did not run or did nothing (nothing added to either `main.js` or `preview.js`
2. `npx storybook add @storybook/addon-vitest` errored when trying to install the browser

**Comment by domyen** [comment]
+1 this issue came up in a live demo we did for the community (a few hundred people)

**Comment by ghengeveld** [comment]
This issue is unrelated to the testing module. The actual problem is that a11y tests don't rerun on HMR (source file update). Fixing the a11y violation implies updating the source file, which should rerun a11y tests to validate the change. There is no need to rerun when Vitest finishes (or testing module results change) because by definition nothing should've changed at that point.
